### PR TITLE
Consolidate GCD codec tests (Follow up to #53149)

### PR DESCRIPTION
**Pull request by rschu1ze**
Follow up to #53149: minor code changes \+ moves all tests into a single place.

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-ch-test-poll2** [comment]
This is an automated comment for commit d36edea9e2da0c5e42a0b269fd6de576b4419378 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/54332/d36edea9e2da0c5e42a0b269fd6de576b4419378/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

File: docs/en/sql-reference/statements/create/table.md [hunk 1]
```diff
@@ -391,19 +391,19 @@ DEFLATE_QPL is not available in ClickHouse Cloud.
...
 
 #### GCD
 
-`GCD()` - - Calculates the greatest common denominator (GCD) of the values in the column, then divides each value by the GCD. Can be used with integer, decimal and date/time columns. A viable use case are timestamps or monetary values with high precision. GCD is a data preparation codec, i.e. cannot be used stand-alone.
+`GCD()` - - Calculates the greatest common denominator (GCD) of the values in the column, then divides each value by the GCD. Can be used with integer, decimal and date/time columns. A viable use case are values which change (increase or decrease) in multiples of the GCD, e.g. 24, 28, 16, 24, 8, 24 (GCD = 4). GCD is a data preparation codec, i.e. it cannot be used stand-alone.
```
Comment (yakov-olkhovskiy) [inline docs/en/sql-reference/statements/create/table.md:406]:
```suggestion
`GCD()` - - Calculates the greatest common denominator (GCD) of the values in the column, then divides each value by the GCD. Can be used with integer, decimal and date/time columns. A viable use case is a values which change (increase or decrease) in multiples of the GCD, e.g. 24, 28, 16, 24, 8, 24 (GCD = 4). GCD is a data preparation codec, i.e. it cannot be used stand-alone.
```

File: tests/queries/0_stateless/00700_decimal_with_default_precision_and_scale.sql [hunk 2]
```diff
@@ -12,7 +12,7 @@ ORDER BY (d2, d3);
 
 INSERT INTO decimal (d1, d2, d3) VALUES (4.2, 4.2, 4.2);
 
-SELECT type FROM system.columns WHERE table = 'decimal' AND database = currentDatabase();
```
[the remark this answers is not in the document]
Comment (rschu1ze) [reply tests/queries/0_stateless/00700_decimal_with_default_precision_and_scale.sql:15]: Missed that, thanks for noticing, fixed.

File: docs/en/sql-reference/statements/create/table.md (hunk 1 above)
Comment (rschu1ze) [reply docs/en/sql-reference/statements/create/table.md:406]: Plural `are values which change` imho sounds a bit better than singular `is a values which change`.

Comment (yakov-olkhovskiy) [reply docs/en/sql-reference/statements/create/table.md:406]: but actually it's a singular "use case" which dictates which verb we are using here... though I'm not a native english speaker, so... :)

**Review by yakov-olkhovskiy — APPROVED** [review]

**Pull request merged**